datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -196,7 +196,14 @@ impl ExecutionPlan for CoalesceBatchesExec {
     }
 
     fn statistics(&self) -> Result<Statistics> {
-        Statistics::with_fetch(self.input.statistics()?, self.schema(), self.fetch, 0, 1)
+        self.partition_statistics(None)
+    }
+
+    fn partition_statistics(&self, partition: Option<usize>) -> Result<Statistics> {
+        let input_stats = self.input.partition_statistics(partition)?;
+        let fetched_stat =
+            Statistics::with_fetch(input_stats.clone(), self.schema(), self.fetch, 0, 1)?;
+        Ok(fetched_stat)
```

Comment:
I think you can avoid a clone here. For example:

```suggestion
        self.input.partition_statistics(partition)?.with_fetch(
            self.schema(),
            self.fetch,
            0,
            1,
        )
```


(This is the pattern you used elsewhere too like in CoalescePartitionsExec)

Reply:
Yes, the clone is unnecessary (It was not even detected by Clippy)